sympy/sets/sets.py

```diff
@@ -916,6 +916,11 @@ def _complement(self, other):
             b = Interval(self.end, S.Infinity, not self.right_open, True)
             return Union(a, b)
 
+        if isinstance(other, FiniteSet):
```

Comment:
@hargup see [this](https://github.com/sympy/sympy/pull/9682/files#diff-e21906560346d439e577d8c1be6b3eceR882)